### DISCUSSION TITLE
分享一个“UseIPv6”情况下疑似会泄露真实IP情况（非bug）

### DISCUSSION THREAD

**Discussion by gissily** [v2fly/v2ray-core#875] [Show and tell]
因为使用vultr提供ipv6，所以部署时候特意设定不同的inbound和outbound，以方便日后出访时候使用ipv4或ipv6，整体的线路图大概如下。
```
browser --socks5-->  client-v2ray --ws--> cloudflare --> nginx -->  vps-v2ray --> internet
```
这是一个日常使用的线路，其中client-v2ray不是桌面版，和vps上的一致目前版本是v4.36.2；在验证这个情况时候，试过跳过cloudflare，直接用ipv4或ipv6直连，结果是只要使用“UseIPv6”，情况都会一样出现疑似泄露情况。

浏览器设置是禁用定位和禁用WebRTC；连接socks5方式尝试过浏览器插件，系统代理；google账号不登入，不存在google账户活动记录。

这个疑似情况目前发现只是会在google搜索结果页出现，就是页面最底下地理位置相关信息，而且只会在以ipv6出访时候出现。

简单说一下操作，以”what is my ip“ 为搜索词条进行搜索，在”UseIPv4“ 时候，搜索结果页面第一条记录上方会显示vps的ipv4地址，页面底部位置信息会显示，”机房地区 - 基于您的 IP 地址“，像我的是日本，具体就是这样”日本 - 基于您的 IP 地址“；如果切换成“UseIPv6” 搜索结果页面第一条记录上方会显示vps的ipv6地址，页面底部位置信息会显示”XX区 中国XX省XX市 - 基于您的 IP 地址“。
在刚发现这个情况，尝试过一堆ip adress服务的网站，全部都是识别vps的ipv6和ipv4；如果结合上述搜索结果页面情况，个人认为以IPv6访问google进行搜索出现”XX区 中国XX省XX市“情况和google页面有关，所以尝试过debug google搜索结果页面，鉴于前端能力有限，未能查出端倪，所以在这分享一下这个情况，希望有大神说明。

**Comment by jimsmt** [comment]
先换个浏览器用隐身模式看能不能复现。google不登陆也记录cookies，前段时间很多人的ip都被google识别成中国了

**Reply by gissily** [reply]
几大浏览器情况一样，只要是用ipv6出去，google就显示我真实的省市

**Comment by k79e** [comment]
我也有这个情况 是个us的v6地址 google还说我中国呢
然后他还说什么我一个美国地址 是阿联酋的 .....

**Comment by okudayukiko** [comment]
我用WireGuard（使用IP4+IP6 NAT），打開https://whatismyipaddress.com 顯示的是伺服器端IPv6位址。

**Reply by gissily** [reply]
这类API方式的网站都是正确显示vps的IPv6及相关位置信息，我是在怀疑是否google的页面的问题；如果方便想用你的配置验证下，用IPv6出访，浏览器禁用定位和禁用WebRTC等相关设置时候，如果浏览器已登录google换成private方式或其他未登录的浏览器访问，google搜索结果页面底下位置信息，如果在“基于您的 IP 地址”情况下，是显示你目前真实地区信息还是vps的地区信息。

**Reply by jimsmt** [reply]
google根据你或者其他使用这个ip的人之前的使用记录，已经将这个ip的地理位置准确识别到中国了

**Comment by okudayukiko** [comment]
應該是發生在，手機App沒有設定好路由表的情況下。
以WireGuard為例：
`Address=10.10.10.2,fd00:10:10::2`
`AllowedIPs=0.0.0.0/0,::/0`
用https://test-ipv6.com 測試，顯示VPS的IPv4位址和IPv6位址。
`Address=10.10.10.2,fd00:10:10::2`
`AllowedIPs=0.0.0.0/0`
用https://test-ipv6.com 測試，顯示VPS的IPv4位址，和ISP的IPv6位址。
`Address=10.10.10.2,fd00:10:10::2`
`AllowedIPs=0.0.0.0/0,2000::/3`
顯示VPS的IPv4位址和IPv6位址。
